src/shared/components/Workspace/Queries/Query/QueryComponent.tsx

```diff
@@ -0,0 +1,178 @@
+import * as React from 'react';
+import { List } from '../../../../store/reducers/lists';
+import './query-component.less';
+import InfiniteScroll from '../../../Animations/InfiniteScroll';
+import ListItem from '../../../Animations/ListItem';
+import { FetchableState } from '../../../../store/reducers/utils';
+import { PaginatedList, Resource } from '@bbp/nexus-sdk';
+import ResourceMetadataCard from '../../../Resources/MetadataCard';
+import { Popover, Icon, Tooltip, Button } from 'antd';
+import TypesIconList from '../../../Types/TypesIcon';
+import RenameableItem from '../../../Renameable';
+import FullTextSearch from './Search';
+import TypesFilter from './Types';
+import SchemasFilter from './Schemas';
+
+const MOUSE_ENTER_DELAY = 0.5;
+
+interface QueryComponentProps {
+  list: List;
+  goToResource: (resource: Resource) => void;
+  goToQuery: (list: List) => void;
+  next: VoidFunction;
+  updateList: (list: List) => void;
+  deleteList: () => void;
+  cloneList: (list: List) => void;
+  handleRefreshList: () => void;
+}
+
+const QueryComponent: React.FunctionComponent<QueryComponentProps> = props => {
+  const {
+    list: {
+      name,
+      query,
+      results: { error, isFetching, data },
+    },
+    handleRefreshList,
+    goToResource,
+    goToQuery,
+    updateList,
+    deleteList,
+    cloneList,
+    next,
+  } = props;
+  const handleOnClick = (resource: Resource) => () => goToResource(resource);
+  const handleUpdate = (value: string) => {
+    updateList({ ...props.list, name: value });
+  };
+  const handleOnSearch = (value: string) => {
+    // update TextQuery value
+    updateList({
+      ...props.list,
+      query: {
+        ...props.list.query,
+        textQuery: value,
+      },
+    });
+  };
+  const handleDelete = () => {
+    deleteList();
+  };
+  const handleClear = () => {
+    // remove the filters
+    // and the textQuery
+    updateList({
+      ...props.list,
+      query: {
+        filters: {},
+      },
+    });
+  };
+
+  const handleCloneList = () => {
+    cloneList({ ...props.list });
+  };
+
+  const handleFilterChange = (value: { [filterKey: string]: string }) => {
+    updateList({
+      ...props.list,
+      query: {
+        ...props.list.query,
+        filters: {
+          ...props.list.query.filters,
+          ...value,
+        },
+      },
+    });
+  };
+
+  const fetchablePaginatedList: FetchableState<PaginatedList<Resource>> = {
+    error,
+    isFetching,
+    data: (data && data.resources) || null,
+  };
+  return (
+    <div className="query-component">
+      <h3 className="header">
+        <RenameableItem
+          defaultValue={`${name} ${isFetching ? 'loading' : ''}`}
+          onChange={handleUpdate}
+          size="small"
+        />
+        <div className="count">
+          {fetchablePaginatedList.data &&
```

Comment:
Could be much more concise if there were a line before the `return` with `const total = fetchablePaginatedList.data && fetchablePaginatedList.data.total || 0;`